### PR TITLE
`tmpnet`: Separate node into orchestration, config and process 

### PR DESCRIPTION
**Pull request by marun**
- Previously tmpnet/node.go contained methods to orchestrate nodes, manage their processes, and read and write their configuration. This PR moves reading/writing of configuration to node_config.go and process management to node_process.go. The separation is intended to improve maintainability.

- A new `NodeRuntime` interface is added to support future pod-based implementation.

### CODE REVIEW

File: tests/fixture/tmpnet/node_process.go [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+			return false, nil
+		}
+	case syscall.Errno:
+		if t == syscall.ECONNREFUSED {
+			// Connection refused - potentially recoverable
+			return false, nil
+		}
+	}
+	// Assume all other errors are not recoverable
+	return false, fmt.Errorf("failed to query node health: %w", err)
```
Comment (danlaine) [inline tests/fixture/tmpnet/node_process.go:53]: In general in avalanchego we don't check whether network errors are potentially temporary/recoverable, and just assume they're not. I'd be ok doing the same here

File: tests/fixture/tmpnet/node_process.go [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+		case <-ticker.C:
+		}
+	}
+	return nil
+}
+
+// Retrieve the node process if it is running. As part of determining
+// process liveness, the node's process context will be refreshed if
+// live or cleared if not running.
+func (p *NodeProcess) getProcess() (*os.Process, error) {
```
Comment (danlaine) [inline tests/fixture/tmpnet/node_process.go:229]: When we do `exec.Command` to start the binary, we have a reference to the `Process`. I think we can use that instead, right?

File: tests/fixture/tmpnet/README.md [hunk 3]
```diff
@@ -31,6 +31,8 @@ the following non-test files:
 | genesis.go        |             | Creates test genesis                           |
 | network.go        | Network     | Orchestrates and configures temporary networks |
 | node.go           | Node        | Orchestrates and configures nodes              |
+| node_config.go    | Node        | Reads and writes node configuration            |
+| node_process.go   | NodeProcess | Orchestrates node processes                    |
```
Comment (abi87) [inline tests/fixture/tmpnet/README.md:35]: line 107: `DefaultRuntime:` should be renamed to `NodeRuntimeConfig` I believe

File: tests/fixture/tmpnet/node_process.go [hunk 4]
```diff
@@ -0,0 +1,258 @@
...
+// Start waits for the process context to be written which
+// indicates that the node will be accepting connections on
+// its staking port. The network will start faster with this
+// synchronization due to the avoidance of exponential backoff
+// if a node tries to connect to a beacon that is not ready.
+func (p *NodeProcess) Start(w io.Writer) error {
+	// Avoid attempting to start an already running node.
+	proc, err := p.getProcess()
+	if err != nil {
+		return fmt.Errorf("failed to start node process: %w", err)
```
Comment (abi87) [inline tests/fixture/tmpnet/node_process.go:100]: nit: this error message seems a bit off. It's not really starting a process here, just checking if it's running already right?

File: tests/fixture/tmpnet/node_process.go [hunk 5]
```diff
@@ -0,0 +1,258 @@
...
+
+	_, err = fmt.Fprintf(w, "Started %s\n", nodeDescription)
+	return err
+}
+
+// Signals the node process to stop.
+func (p *NodeProcess) InitiateStop() error {
+	proc, err := p.getProcess()
+	if err != nil {
+		return fmt.Errorf("failed to retrieve process to stop: %w", err)
```
Comment (abi87) [inline tests/fixture/tmpnet/node_process.go:153]: this is a better error message. Consider adapting it for p.Start as well

**Review by abi87 — APPROVED** [review]
found minor nits but LGTM overall

**Review by danlaine — APPROVED** [review]

**Pull request merged**